src/manifolds/Euclidean.jl

```diff
@@ -231,7 +237,7 @@ end
 Return the array dimensions required to represent an element on the
 [`Euclidean`](@ref) `M`, i.e. the vector of all array dimensions.
 """
-@generated representation_size(::Euclidean{N}) where N = Tuple(N.parameters...)
+Base.@pure representation_size(::Euclidean{N}) where {N} = tuple(N.parameters...)
```

Comment:
As far as I understand `Base.@pure`, it's supposed to refer to the entire function, not just one method. So this is incorrect.

Reply:
Not quite. It can be applied to a single method, but once a `Base.@pure` method for a function has been called then no new methods may be added (I think; see [this discussion](https://discourse.julialang.org/t/pure-macro/3871/2)). Either way, as written here this is dangerous, so I'll fix it.

Reply:
Looks like we already have `size_to_tuple` for this, so we can just replace it.

Reply:
Yes, either way it's incorrect to use `@pure` here.